File: 804-unique-morse-code-words/804-unique-morse-code-words.cpp
class Solution {
public:
    int uniqueMorseRepresentations(vector<string>& words) {
        vector<string> morse{".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        
        set<string> code;
        for(string w: words)
        {
            string sb="";
            for(char c:w)
            {
                sb+=morse[c-'a'];
            }
            code.insert(sb);
        }
            
        return code.size();
    }
};

File: 804-unique-morse-code-words/804-unique-morse-code-words.java
class Solution {
    public int uniqueMorseRepresentations(String[] words) {
        String []morse=new String[]{".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        
        Set<String> code=new HashSet<>();
        for(String w: words)
        {
            StringBuilder sb=new StringBuilder();
            for(char c:w.toCharArray())
            {
                sb.append(morse[c-'a']);
            }
            code.add(sb.toString());
        }
            
        return code.size();
    }
}